app/src/main/java/com/example/androidtrainingproject/ui/shared/AppTopBar.kt

```diff
@@ -47,15 +55,36 @@ fun AppTopBar(
             navigationIconContentColor = Black
         ),
         actions = {
-            if (accountIcon == true) {
-                IconButton(onClick = {}) {
+            if (accountIcon) {
+                IconButton(onClick = { destinationsNavigator.navigate(UserProfileDestination) }) {
                     Icon(Icons.Default.AccountCircle, stringResource(id = R.string.account_box_description))
                 }
             }
-            if (cartIcon == true) {
-                IconButton(onClick = {}) {
-                    Icon(Icons.Filled.ShoppingCart, stringResource(id = R.string.shopping_cart_description))
+            if (cartIcon) {
+                Box {
+                    if (!GlobalStorage.cartProducts.isEmpty()) {
+                        Text(
```

Comment:
Feels like I'm seeing the same `Text` over and over again throughout the app. If that's the case, please extract it as a shared (as in a `shared` folder) reusable composable.